angular.module("security.retryQueue",[]).factory("securityRetryQueue",["$q","$log",function(e,r){var n=[],t={onItemAddedCallbacks:[],hasMore:function(){return n.length>0},push:function(e){n.push(e),angular.forEach(t.onItemAddedCallbacks,function(n){try{n(e)}catch(e){r.error("securityRetryQueue.push(retryItem): callback threw an error"+e)}})},pushRetryFn:function(r,n){1===arguments.length&&(n=r,r=void 0);var u=e.defer(),o={reason:r,retry:function(){e.when(n()).then(function(e){u.resolve(e)},function(e){u.reject(e)})},cancel:function(){u.reject()}};return t.push(o),u.promise},retryReason:function(){return t.hasMore()&&n[0].reason},cancelAll:function(){for(;t.hasMore();)n.shift().cancel()},retryAll:function(){for(;t.hasMore();)n.shift().retry()}};return t}]);